src/logic/operate.js

```diff
@@ -0,0 +1,34 @@
+import Big from 'big.js';
+
+const operate = (number1, number2, operator) => {
+  if (!number1 || !number2 || !operator) return '0';
+
+  let res = 0;
+  const num1 = typeof number1 !== 'number' ? Big(Number(number1)) : Big(number1);
+  const num2 = typeof number2 !== 'number' ? Big(Number(number2)) : Big(number2);
+
+  switch (operator) {
+    case '+':
+      res = num1.plus(num2);
+      break;
+    case '-':
+      res = num1.minus(num2);
+      break;
+    case 'x':
+      res = num1.times(num2);
+      break;
+    case '/':
+      res = num1.div(num2);
+      break;
+    case '%':
+      res = num1.times(num2).div(100);
```

Comment:
- This logic doesn't make sense. Why are you multiplying the two numbers? 
- Simply do: `res = num1.mod(num2)`